crates/orderbook/openapi.yml

```diff
@@ -469,6 +469,13 @@ components:
       description: 20 byte Ethereum address encoded as a hex with `0x` prefix.
       type: string
       example: "0x6810e776880c02933d47db1b9fc05908e5386b96"
+    AppData:
+      description: |
+        The string encoding of a JSON object representing some `appData`. The
```

Comment:
nanonit: since the field, which is a hash, is called appData maybe we can be super clear here that this is expecting the pre-image

Reply:
The field `AppData`, in the context of the OrderCreation or OrderRequest can be a hash (for backward compatibility) or the stringified JSON representation of the app data doc.
The field `AppDataHash` itself can only be what you describe.

Not sure it's what you mean, but I did consider the names confusing.

For example, there are 3 `AppData*` fields:

``` yaml
    AppData:
      description: |
        The string encoding of a JSON object representing some `appData`. The
        format of the JSON expected in the `appData` field is defined
        [here](https://github.com/cowprotocol/app-data).
      type: string
      example: "{\"version\":\"0.9.0\",\"metadata\":{}}"
    AppDataHash:
      description: 32 bytes encoded as hex with `0x` prefix.
      type: string
      example: "0x0000000000000000000000000000000000000000000000000000000000000000"
    AppDataDocument:
      description: An `appData` document that is registered with the API.
      type: object
      properties:
        fullAppData:
          $ref: "#/components/schemas/AppData"
      required:
        - appData
```

I would prefer to name what's currently called `AppData` -> `AppDataDocument`, but that name is already taken.
Thinking about it again, it makes sense to rename it to something else and use it for the document. I'll update.

Unless you mean something else, in which case, please let me know.